lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -652,18 +652,19 @@ def flatten_array(flattened, array, prefix)
         end
       end
 
-      def headers(options = {})
-        key = options[:key] || @api_key
-        idempotency_key = options[:idempotency_key]
+      def key(options = {})
+        options[:key] || @api_key
+      end
 
+      def headers(options = {})
         headers = {
-          'Authorization' => 'Basic ' + Base64.strict_encode64(key.to_s + ':').strip,
+          'Authorization' => 'Basic ' + Base64.strict_encode64(key(options).to_s + ':').strip,
```

Comment:
will `.to_s` here crash if `@api_key` is `nil`? I see that this isn't a new issue here, so maybe not worth blocking this time sensitive change?